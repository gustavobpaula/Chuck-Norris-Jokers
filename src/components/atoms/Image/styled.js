import styled from 'styled-components';

export const Image = styled.img`
  height: ${props => props.height};
  max-height: ${props => props.maxHeight};
  max-width: ${props => props.maxWidth};
  width: ${props => props.width};
`;
