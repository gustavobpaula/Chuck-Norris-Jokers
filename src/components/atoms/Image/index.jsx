import React from 'react';
import PropTypes from 'prop-types';
import * as Styled from './styled';

const Image = ({ src, alt, height, width, maxHeight, maxWidth, ...rest }) => (
  <Styled.Image
    src={src}
    alt={alt}
    height={height}
    maxHeight={maxHeight}
    maxWidth={maxWidth}
    width={width}
    {...rest}
  />
);

Image.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  height: PropTypes.string,
  width: PropTypes.string,
  maxHeight: PropTypes.string,
  maxWidth: PropTypes.string,
};

Image.defaultProps = {
  maxHeight: '100%',
  maxWidth: '100%',
  height: 'auto',
  width: 'auto',
};

export default Image;
