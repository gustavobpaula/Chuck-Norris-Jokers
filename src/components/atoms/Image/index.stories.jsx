import React from 'react';
import Image from './index';
import chuckNorris from '../../../assets/images/chuck-norris.svg';

export default {
  component: Image,
  title: 'Atoms/Image',
};

export const element = () => <Image alt="Chuck Norris" src={chuckNorris} />;

element.story = {
  name: 'Default',
};

export const customSizes = () => (
  <Image alt="Chuck Norris" src={chuckNorris} height="200px" />
);

customSizes.story = {
  name: 'Custom Sizes',
};
