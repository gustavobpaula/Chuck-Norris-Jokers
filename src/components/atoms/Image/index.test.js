import React from 'react';
import { mount } from 'enzyme';
import chuckNorris from '../../../assets/images/chuck-norris.svg';
import Image from './index';

describe('<Image />', () => {
  it('should mount properly', async () => {
    const wrapper = mount(<Image alt="Chuck Norris" src={chuckNorris} />);

    expect(wrapper).toMatchSnapshot();
  });
});
