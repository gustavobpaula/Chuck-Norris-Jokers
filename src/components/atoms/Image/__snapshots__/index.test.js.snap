// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should mount properly 1`] = `
<Image
  alt="Chuck Norris"
  height="auto"
  maxHeight="100%"
  maxWidth="100%"
  src="test-file-stub"
  width="auto"
>
  <style__Image
    alt="Chuck Norris"
    height="auto"
    maxHeight="100%"
    maxWidth="100%"
    src="test-file-stub"
    width="auto"
  >
    <StyledComponent
      alt="Chuck Norris"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "style__Image-zsbil-0",
            "isStatic": false,
            "lastClassName": "bSPXyq",
            "rules": Array [
              "height:",
              [Function],
              ";max-height:",
              [Function],
              ";max-width:",
              [Function],
              ";width:",
              [Function],
              ";",
            ],
          },
          "displayName": "style__Image",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "style__Image-zsbil-0",
          "target": "img",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      height="auto"
      maxHeight="100%"
      maxWidth="100%"
      src="test-file-stub"
      width="auto"
    >
      <img
        alt="Chuck Norris"
        className="style__Image-zsbil-0 bSPXyq"
        height="auto"
        src="test-file-stub"
        width="auto"
      />
    </StyledComponent>
  </style__Image>
</Image>
`;
